vaadin-radio-button-flow-integration-tests/src/main/java/com/vaadin/flow/component/radiobutton/tests/RadioButtonGroupDemoPage.java

```diff
@@ -0,0 +1,356 @@
+
+        group.setId("button-group-disabled");
+
+        addCard("Disabled radio button group", group);
+    }
+
+    private void addReadOnlyGroup() {
+        // begin-source-example
+        // source-example-heading: Read-only radio button group
+        Div valueInfo = new Div();
+
+        RadioButtonGroup<String> group = new RadioButtonGroup<>();
+        group.setItems("foo", "bar", "baz");
+        group.setReadOnly(true);
+
+        NativeButton button = new NativeButton("Switch read-only state",
+                event -> group.setReadOnly(!group.isReadOnly()));
+        group.addValueChangeListener(
+                event -> valueInfo.setText(group.getValue()));
+        // end-source-example
+
+        group.setId("button-group-read-only");
+        valueInfo.setId("selected-value-info");
+        button.setId("switch-read-only");
+
+        addCard("Read-only radio button group", group, button, valueInfo);
+    }
+
+    private void addDisabledItems() {
+
+        Div valueInfo = new Div();
+        // begin-source-example
+        // source-example-heading: Radio button group with item enabled provider
+        RadioButtonGroup<String> group = new RadioButtonGroup<>();
+        group.setItems("foo", "bar", "baz");
+        group.setItemEnabledProvider(item -> !"bar".equals(item));
+        // end-source-example
+
+        group.addValueChangeListener(
+                event -> valueInfo.setText(group.getValue()));
+
+        group.setId("button-group-disabled-items");
+        valueInfo.setId("button-group-disabled-items-info");
+
+        addCard("Radio button group with item enabled provider", group,
+                valueInfo);
+    }
+
+    private String getName(Person person) {
+        if (person == null) {
+            return null;
+        }
+        return person.getName();
+    }
+
+    private void addComponentAfterItems() {
+        // begin-source-example
+        // source-example-heading: Add component to group
+        RadioButtonGroup<String> group = new RadioButtonGroup<>();
+        group.setItems("foo", "bar", "baz");
+        group.add(new Label("My Custom text"));
+
+        group.getElement().getStyle().set("display", "flex");
+        group.getElement().getStyle().set("flexDirection", "column");
+        // end-source-example
+
+        group.setId("button-group-with-appended-text");
+
+        addCard("Add component to group", group);
+    }
+
+    private void insertComponentsBetweenItems() {
+        // begin-source-example
+        // source-example-heading: Insert component after item in group
+        RadioButtonGroup<String> group = new RadioButtonGroup<>();
+
+        // Note that setting items clear any components
+        group.add(new Label("Foo group"), getFullSizeHr());
+
+        group.setItems("foo", "bar", "baz");
+        group.addComponents("foo", new Label("Not foo selections"),
+                getFullSizeHr());
+
+        group.getElement().getStyle().set("display", "flex");
```

Comment:
The demos shouldn't need to use element-level style tweaks.

Reply:
These are the old demos. I just moved them to the 'tests' part. Should I change it?

Reply:
Yes, the demos should not have this kind of hacks

Reply:
Ah, didn't understand the "tests" part correctly. So they're just moved from demos as integration tests and the new demos no longer use this code. Although it's weird to have this code in tests as well, it's not that important to fix anymore.